truss/base/truss_config.py

```diff
@@ -154,12 +155,70 @@ def to_list(self, verbose=False) -> List[Dict[str, str]]:
         return [model.to_dict(verbose=verbose) for model in self.models]
 
 
+@dataclass
+class HealthCheck:
+    restart_check_delay_seconds: int = 0
+    restart_failure_threshold_seconds: int = MAX_FAILURE_THRESHOLD_SECONDS
+    stop_traffic_failure_threshold_seconds: int = MAX_FAILURE_THRESHOLD_SECONDS
+
+    def __post_init__(self):
```

Comment:
I think we want to do these validations on the backend. That gives us more flexibility to not be stuck with these checks for old clients.

Can you look into how that might work?

Reply:
Discussed offline--I'll use this as an opportunity to create a first version of a `validate_truss_config` GQL mutation that we can call from here: https://github.com/basetenlabs/truss/blob/main/truss/remote/baseten/remote.py#L124

Reply:
I don't understand why we write so much boilerplate code for this here when pydantic can do most of this. - more a comment on pre-existing code though...


I think we should refactor this whole file soon. It's also super error prone that you have to add key literals to `to_dict` methods each time you add a field.